### PR TITLE
Added BillingRoleAssignment, BillingRoleDefinitions and BillingPermissions swagger in GA 2023…

### PR DESCRIPTION
**Pull request by dharapatel27**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [X] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_
  
## Changelog

**BillingRoleAssignments**
1. Changed the customer scope to BillingAccounts/BillingProfiles/Customers scope
2.  Added Support of these APIs for EA

**BillingPermissions**
1. Added Check Access APIs in GA
2. Changed the customer scope to BillingAccounts/BillingProfiles/Customers scope
3. Added Support of these APIs for EA

**BillingRoleDefinitions**
1. Changed the customer scope to BillingAccounts/BillingProfiles/Customers scope
2.  Added Support of these APIs for EA

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingPermission.json [hunk 1]
```diff
@@ -1,55 +1,53 @@
 {
   "swagger": "2.0",
   "info": {
-    "description": "Billing client provides access to billing resources for Azure subscriptions.",
     "title": "BillingManagementClient",
+    "description": "Documentation for Microsoft.Billing.",
```
Comment (dharapatel27) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingPermission.json:5]: to be updated after checking w PMs.

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingPermission.json [hunk 2]
```diff
@@ -1,55 +1,53 @@
...
+    "contact": {
+      "name": "Microsoft.Billing",
+      "url": "https://learn.microsoft.com/en-us/rest/api/billing/",
+      "email": "cabbpt@microsoft.com"
```
Comment (dharapatel27) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingPermission.json:9]: to be updated after checking w PMs.

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingPermission.json [hunk 3]
```diff
@@ -1,55 +1,53 @@
...
-  "schemes": [
-    "https"
-  ],
-  "consumes": [
-    "application/json"
-  ],
-  "produces": [
-    "application/json"
-  ],
   "paths": {
```
Comment (dharapatel27) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingPermission.json:13]: /checkAccess should also be part of this GA

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 4]
```diff
@@ -1,517 +1,1962 @@
...
-  "produces": [
-    "application/json"
-  ],
   "paths": {
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/billingProfiles/{billingProfileName}/billingRoleAssignments/{billingRoleAssignmentName}": {
+      "delete": {
+        "tags": [
+          "BillingRoleAssignment"
+        ],
+        "description": "Deletes a role assignment for the caller on a billing profile. The operation is supported for billing accounts with agreement type Microsoft Partner Agreement or Microsoft Customer Agreement.",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:19]: work with PM to rephrase it

Comment (ramkd12) [reply specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:19]: Isn't EA supported?

Comment (ramkd12) [reply specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:19]: Applicable comments for all APIs below

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 5]
```diff
@@ -1,517 +1,1962 @@
...
+            "$ref": "./types.json#/parameters/billingProfileName"
+          },
+          {
+            "$ref": "./types.json#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "./types.json#/parameters/filter"
+          },
+          {
+            "in": "query",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:145]: why is expand needed here in first place? what does it do?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 6]
```diff
@@ -1,517 +1,1962 @@
...
+            "name": "expand",
+            "description": "May be used to expand the participants.",
+            "type": "string"
+          },
+          {
+            "$ref": "./types.json#/parameters/top"
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:151]: are we going to support top, filter, skip?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 7]
```diff
@@ -1,517 +1,1962 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/billingProfiles/{billingProfileName}/createBillingRoleAssignment": {
+      "post": {
+        "tags": [
+          "BillingRoleAssignment"
+        ],
+        "description": "Adds a role assignment on a billing profile. The operation is supported for billing accounts with agreement type Microsoft Customer Agreement.",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:186]: why not MPA?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 8]
```diff
@@ -1,517 +1,1962 @@
...
+          "BillingRoleAssignmentListByBillingProfile": {
+            "$ref": "./examples/billingRoleAssignmentListByBillingProfile.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/billingProfiles/{billingProfileName}/createBillingRoleAssignment": {
```
Comment (dharapatel27) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:181]: follow up with AuthZ on if PUT can be supported.

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 9]
```diff
@@ -1,517 +1,1962 @@
...
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "202": {
+            "description": "The properties of the role assignment.",
+            "schema": {
+              "$ref": "#/definitions/BillingRoleAssignment"
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:231]: Shouldn't this return 202 with LocationHeader?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 10]
```diff
@@ -1,517 +1,1962 @@
...
+            }
+          },
+          "default": {
+            "description": "Default response",
+            "schema": {
+              "$ref": "./types.json#/definitions/ArmErrorResponse"
```
Comment (dharapatel27) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:237]: check if this only should be a location header

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 11]
```diff
@@ -1,517 +1,1962 @@
...
+          },
+          {
+            "$ref": "./types.json#/parameters/billingProfileName"
+          },
+          {
+            "$ref": "./types.json#/parameters/customerName"
+          },
+          {
+            "in": "query",
+            "name": "enablePagination",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:384]: get rid of this parameter.

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 12]
```diff
@@ -1,517 +1,1962 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "The list of role assignments.",
+            "schema": {
+              "$ref": "#/definitions/BillingRoleAssignmentListResult"
+            }
+          },
+          "202": {
+            "description": "The list of role assignments.",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:552]: same as earlier 202 comment reg Location header

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 13]
```diff
@@ -1,517 +1,1962 @@
...
+          },
+          {
+            "$ref": "./types.json#/parameters/billingProfileName"
+          },
+          {
+            "$ref": "./types.json#/parameters/invoiceSectionName"
+          },
+          {
+            "in": "query",
+            "name": "enablePagination",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:712]: get rid of it

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 14]
```diff
@@ -1,517 +1,1962 @@
...
+          "BillingRoleAssignmentGetByBillingAccount": {
+            "$ref": "./examples/billingRoleAssignmentGetByBillingAccount.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "BillingRoleAssignment"
+        ],
+        "description": "Create or update a billing role assignment.",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:1072]: specify that this is only for EA

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 15]
```diff
@@ -1,517 +1,1962 @@
...
+        },
+        "operationId": "BillingRoleAssignments_DeleteByDepartment",
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
-            "$ref": "./types.json#/parameters/apiVersionParameter"
+            "in": "path",
+            "name": "billingAccountName",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:1269]: why are these parameters differently formatted?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 16]
```diff
@@ -1,517 +1,1962 @@
...
+          "ResolveBillingRoleAssignmentByBillingAccount": {
+            "$ref": "./examples/resolveBillingRoleAssignmentByBillingAccount.json"
           }
         }
       }
     }
   },
   "definitions": {
     "BillingRoleAssignment": {
+      "description": "Common fields that are returned in the response for all Azure Resource Manager resources.",
```
Comment (dharapatel27) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:1768]: change this to be more descriptive

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json [hunk 17]
```diff
@@ -1,170 +1,299 @@
...
           },
           {
             "$ref": "./types.json#/parameters/billingProfileName"
           },
           {
-            "$ref": "#/parameters/billingRoleDefinitionName"
+            "$ref": "./types.json#/parameters/customerName"
+          },
+          {
+            "in": "path",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json:144]: should be from types.json

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json [hunk 18]
```diff
@@ -1,170 +1,299 @@
...
+            "$ref": "./types.json#/parameters/billingAccountName"
+          },
+          {
+            "$ref": "./types.json#/parameters/billingProfileName"
+          },
+          {
+            "$ref": "./types.json#/parameters/invoiceSectionName"
+          },
+          {
+            "in": "path",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json:251]: should be from types.json

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json [hunk 19]
```diff
@@ -175,227 +304,458 @@
...
+        "operationId": "BillingRoleDefinition_GetByBillingAccount",
+        "produces": [
+          "application/json"
         ],
+        "parameters": [
+          {
+            "$ref": "./types.json#/parameters/billingAccountName"
+          },
+          {
+            "in": "path",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json:352]: should be from types.json

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json [hunk 20]
```diff
@@ -175,227 +304,458 @@
...
+        ],
+        "parameters": [
           {
             "$ref": "./types.json#/parameters/billingAccountName"
+          },
+          {
+            "$ref": "./types.json#/parameters/departmentName"
+          },
+          {
+            "in": "path",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json:450]: should be from types.json

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json [hunk 21]
```diff
@@ -175,227 +304,458 @@
...
           {
-            "$ref": "./types.json#/parameters/apiVersionParameter"
+            "$ref": "./types.json#/parameters/billingAccountName"
           },
           {
-            "$ref": "./types.json#/parameters/billingAccountName"
+            "$ref": "./types.json#/parameters/enrollmentAccountName"
+          },
+          {
+            "in": "path",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json:551]: should be from types.json

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json [hunk 22]
```diff
@@ -175,227 +304,458 @@
...
+          },
+          {
+            "in": "query",
+            "name": "accountName",
+            "description": "The name of the enrollment account.",
+            "type": "string",
+            "pattern": "^[1-9][0-9]*$"
+          },
+          {
+            "in": "path",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json:608]: should be from types.json

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json [hunk 23]
```diff
@@ -175,227 +304,458 @@
...
+        "operationId": "BillingRoleDefinition_ListByEnrollmentAccount",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "./types.json#/parameters/billingAccountName"
+          },
+          {
+            "in": "query",
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleDefinition.json:601]: should be from types.json

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 24]
```diff
@@ -1,517 +1,1962 @@
...
+          },
+          {
+            "$ref": "./types.json#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "./types.json#/parameters/filter"
+          },
+          {
+            "in": "query",
+            "name": "expand",
```
Comment (wilcobmsft) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json:146]: Shouldn't this be "$expand"?

**Review by wilcobmsft — APPROVED** [review]

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json [hunk 25]
```diff
@@ -1,517 +1,1867 @@
...
+          "type": "string"
+        },
+        "modifiedByPrincipalTenantId": {
+          "type": "string"
+        },
+        "notificationContact": {
+          "description": "The Notification Contact person email address",
+          "type": "string"
+        },
+        "notificationFrequency": {
```
[the remark this answers is not in the document]
Comment (himanshuPandey25) [reply specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingRoleAssignment.json]: this is resloved

**Pull request merged**